### ISSUE TITLE
Question: Is classification training on only one task different to normal Torch training?

### ISSUE THREAD

**Issue by Peter-Devine** [snorkel-team/snorkel#1456]
Sorry if this is off-topic, I just did not know where else to ask this question.

I like the Snorkel Classification modules, as they are fairly intuitive to use, and make adding new tasks to an MTL model a cinch.

If I create an MTL model with only **one task**, will that model train in the same was as if I were to train it through vanilla PyTorch? 

From the best of my understanding, I think that it would, but I was wondering if there would be any subtle peculiarities with the training regime of Snorkel's Classification modules.
The reason I would like to use the Snorkel MTL framework to do single task training is that I would later like to easily add additional tasks to training, and would not like to change the set-up too much.

Thanks

**Comment by bhancock8** [comment]
Hi @Peter-UCL, great question! First of all, we're glad you're enjoying the MultitaskClassifier model! Model architectures are not the focus of the Snorkel repo, but we found it so useful in our own prototyping to have a model with that much control that we decided to open source it as well.

Now to answer your question: yes, I believe an MTL model made in Snorkel with only one task will train the exact same a single-task model. There will only be one loss function, that loss will propagate through all the parameters it touches (in this case, all the model parameters), and training will occur as normal.
